api/src/main/java/tennisclub/dto/CourtDto.java

```diff
@@ -0,0 +1,74 @@
+package tennisclub.dto;
+
+import tennisclub.enums.CourtType;
+
+public class CourtDto {
+    private Long id;
+    private String name;
+    private String address;
+    private CourtType type;
+    private String previewImageUrl;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public CourtType getType() {
+        return type;
+    }
+
+    public void setType(CourtType type) {
+        this.type = type;
+    }
+
+    public String getPreviewImageUrl() {
+        return previewImageUrl;
+    }
+
+    public void setPreviewImageUrl(String previewImageUrl) {
+        this.previewImageUrl = previewImageUrl;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof CourtDto)) return false;
+
+        CourtDto courtDto = (CourtDto) o;
+
+        if (getName() != null ? !getName().equals(courtDto.getName()) : courtDto.getName() != null) return false;
```

Comment:
Isn't actually `name` and `address` enough _(as in the entity class)_? 🤔

Reply:
@Fuyunori, `*sigh*` maybe.

- All internet sources suggest that you are right. (I.e. `equals()` should be based on a **small** set of attributes ideally forming a candidate key.) That is what I based my implementation of `CourtDao.equals()` on, as well as my previous comments on (I believe) your past PRs. I also criticized another team in the milestone 1 review for an overblown `equals()`.

-  On the other hand, the seminar's *vox populi* says otherwise. And, ironically, we also got `equals()`-related criticism in the review -- that our `equals()` DAO methods are attribute-wise incomplete. :smile: 

Honestly, I am getting tired of the idiosyncrasies of the Java EE stack and the "religious debates" around it. And I kind of do not care anymore, as long as I get a passing grade, TBH. :frowning_face: 

So... sure, let me *fix* this.

Reply:
@Fuyunori, done.

Reply:
@paveltobias Thank you very much and good job! ❤️